mockgen/mockgen.go

```diff
@@ -622,22 +628,23 @@ func (g *generator) Output() []byte {
 	return src
 }
 
-func lookupPackageName(importPath string) (string, bool) {
-	var pkg struct {
-		Name string
-	}
+func lookupPackagesName(importPaths []string) map[string]string {
+	names := make(map[string]string)
 	b := bytes.NewBuffer(nil)
-	cmd := exec.Command("go", "list", "-json", importPath)
+	args := []string{"list", "-f", "{{.Name}} {{.ImportPath}}"}
+	args = append(args, importPaths...)
+	cmd := exec.Command("go", args...)
 	cmd.Stdout = b
-	err := cmd.Run()
-	if err != nil {
-		return "", false
-	}
-	err = json.Unmarshal(b.Bytes(), &pkg)
-	if err != nil {
-		return "", false
+	cmd.Run()
+	sc := bufio.NewScanner(b)
+	for sc.Scan() {
```

Comment:
can we keep the logic as processing json? I like it better for a couple of reasons:

- Less code
- can use named members vs `pkg[0]`
- gets rid of the quote issue you mentioned in another file.

Reply:
I did it with formatted output because it outputs less data and works a little bit faster, but you are right that json logic is cleaner. Rewrote it.